tiledb/sm/query/readers/ordered_dim_label_reader.cc

```diff
@@ -638,7 +638,11 @@ IndexType OrderedDimLabelReader::search_for_range(
 
   // Run a binary search.
   Op cmp;
```

Comment:
I wrapped the original loop with if (left_index != right_index) { and it seems to be fixing the added test case... Does it also fix your original case?

Reply:
It should. The original case was the result of a typo in a Python examples: If the right side of the tile is 0 on an unsigned dimension, then subtracting 1 from the right side creates a very large number and the end condition is never met. This is only possible if the tile size is 1 (the typo).